src/viewport/s3_service.py

```
@@ -120,6 +120,30 @@ def _get_presign_client(self) -> "S3Client":
+                    "AllowedMethods": ["GET", "PUT", "POST", "HEAD", "DELETE"],
+                    "AllowedHeaders": ["*"],
+                    "ExposeHeaders": ["ETag"],
+                    "MaxAgeSeconds": 3600,
+                }
+            ]
+        }
+        try:
+            async with self._get_s3_client() as s3:
+                await s3.put_bucket_cors(Bucket=self.settings.bucket, CORSConfiguration=cors_config)
+            logger.info("Bucket CORS configured with ETag exposure")
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)